File: check.py
import os
import re

root = r'd:\Website\octg-site'
issues = []

actual_files = {}
for dirpath, dirnames, filenames in os.walk(root):
    for f in filenames:
        full = os.path.join(dirpath, f).replace('\\', '/')
        actual_files[full.lower()] = f

for dirpath, dirnames, filenames in os.walk(root):
    for f in filenames:
        if not f.endswith('.php'): continue
        path = os.path.join(dirpath, f)
        with open(path, 'r', encoding='utf-8') as file:
            content = file.read()
            
        includes = re.findall(r"(?:include|require)(?:_once)?\s*(?:__DIR__\s*\.\s*)?['\"]([^'\"]+)['\"]", content)
        for inc in includes:
            inc_path = inc.lstrip('/')
            if inc.startswith('../'):
                check = os.path.normpath(os.path.join(dirpath, inc))
            else:
                check = os.path.normpath(os.path.join(dirpath, inc_path))
            
            check_lower = check.replace('\\', '/').lower()
            if check_lower in actual_files:
                expected = os.path.basename(inc)
                actual = actual_files[check_lower]
                if expected != actual:
                    issues.append(f'CASE MISMATCH: {inc} in {f} (Actual: {actual})')
            else:
                 issues.append(f'BROKEN INCLUDE: {inc} in {f}')
                 
print("ISSUES:")
for i in issues: print(i)

File: check.php
<?php
$issues = [];
$dir = new RecursiveDirectoryIterator(__DIR__);
$ite = new RecursiveIteratorIterator($dir);
$files = new RegexIterator($ite, '/\.php$/', RegexIterator::GET_MATCH);
foreach($files as $file) {
    $path = $file[0];
    $content = file_get_contents($path);
    if(preg_match_all("/(?:include|require)(?:_once)?\s*(?:__DIR__\s*\.\s*)?['\"]([^'\"]+)['\"]/", $content, $matches)) {
        foreach($matches[1] as $inc) {
            $checkPath = dirname($path) . '/' . $inc;
            if(strpos($inc, '/') === 0) {
                 $checkPath = __DIR__ . $inc;
            }
            if (!file_exists($checkPath)) {
                $issues[] = "BROKEN INCLUDE: $inc in $path (Resolved to: $checkPath)";
            } else {
                // Check case sensitivity
                $real = realpath($checkPath);
                $basename = basename($inc);
                if (basename($real) !== $basename) {
                    $issues[] = "CASE SENSITIVITY: $inc in $path (Actual: " . basename($real) . ")";
                }
            }
        }
    }
}
print_r($issues);
